src/Microsoft.NET.Sdk.Functions.MSBuild/Tasks/BuildFunctions.cs

```diff
@@ -29,8 +31,50 @@ public override bool Execute()
                 File.WriteAllText(hostJsonPath, hostJsonString);
             }
 
-            return FunctionJsonConvert.TryConvert(TargetPath, OutputPath, Log);
+            Assembly taskAssembly = typeof(BuildFunctions).GetTypeInfo().Assembly;
+#if NET46
+            var info = new ProcessStartInfo
+            {
+                UseShellExecute = false,
+                CreateNoWindow = false,
+                RedirectStandardError = true,
+                RedirectStandardOutput = true,
+                WorkingDirectory = Path.GetDirectoryName(taskAssembly.Location),
+                FileName = Path.Combine(Path.GetDirectoryName(taskAssembly.Location), "Microsoft.NET.Sdk.Functions.Console.exe"),
```

Comment:
we may need to move the exe and dll to a common location as well 
i.e
tools\net46\<task>.dll
tools\netcoreapp2.0\<task>.dll
tools\functions\Functions.console.exe
tools\functions\Functions.console.dll

Reply:
why?

Reply:
because the task in net46 & task in netcoreapp2.0 will need to call both .exe and dotnet .dll 

scenarios are:
1. App targeting net46 triggered by desktop msbuild will invoke the task under net46\ & this needs to call console.exe
2. App targeting netstandard triggered by desktop msbuild will invoke the task under net46\ & this needs to call dotnet console.dll

3. App targeting net46 triggered by core msbuild will invoke the task under netcoreapp2.0\ & this needs to call console.exe
4. App targeting netstandard triggered by core msbuild will invoke the task under netcoreapp2.0\ & this needs to call dotnet console.dll

Reply:
@vijayrkn for what the generator is currently doing, it should be fine to use the exe (full framework) app when running under MSBuild and the netstandard version when running on core. The full framework console app should be able to load and inspect the .NET Standard assemblies and vice-versa (as long as the full framework console app is properly referencing .NET Standard). What are your concerns with this approach? Am I incorrect here?

Reply:
@fabiocav - Thanks for confirming. As long as generator.exe can inspect and load both netfx app and netcoreapp and vice versa (dotnet generator.dll can inspect and load both netfx and netcoreapp), we should be good.